src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java

```diff
@@ -362,23 +369,50 @@ private PaymentTransactionInfoPlugin executeInitialTransaction(final Transaction
                                                                    final Currency currency,
                                                                    final Iterable<PluginProperty> properties,
                                                                    final CallContext context) throws PaymentPluginApiException {
-        final Account account = getAccount(kbAccountId, context);
-
-        final Long transactionAmount = (amount == null ? null : amount.longValue());
-        final PaymentData paymentData = buildPaymentData(account, kbPaymentId, kbTransactionId, kbPaymentMethodId, currency, properties, context);
-        final OrderData orderData = buildOrderData(account, properties);
-        final UserData userData = buildUserData(account, properties);
-        final String termUrl = PluginProperties.findPluginPropertyValue(PROPERTY_TERM_URL, properties);
-        final SplitSettlementData splitSettlementData = null;
-        final DateTime utcNow = clock.getUTCNow();
+        killbillMetrics.markMeter(transactionType.toString());
```

Comment:
Completely agree we need to hide the metrics glue in the [base](https://github.com/killbill/killbill-plugin-framework-java) plugin. I think there are at least 2 options we could consider:
1. We do the glue in the base plugin with metrics and define an abstract method that would be implemented in the plugin itself with the real code
2. We provide in the base plugin a method like `withMetrics(OperationCallback cb)` which would wrap the callback wtih the metrics and where `OperationCallback` is implemented in the plugin itself and contains the real code. 

I am actually more inclined towards solution 2.